python/test/regression/test_cast_matmul.py

```diff
@@ -94,4 +94,9 @@ def matmul_kernel(A, B, C, M, N, K,  #
         BLOCK_N=BLOCK_N,  #
         BLOCK_K=BLOCK_K)
 
-    torch.testing.assert_close(out_torch, out_triton, atol=0.3, rtol=0.01)
+    # Torch can compute reference result on CPU using fp32 arithmetics for fp16
+    # test. Such reference requires increased tolerance for big K values.
+    if out_dtype == "float16" and K > 128:
+        torch.testing.assert_close(out_torch, out_triton, atol=2, rtol=0.1)
```

Comment:
that's a pretty big difference. Is it because the matmul is computed on the CPU instead of the GPU? What would be the tolerance if we ran trough numpy?